File: src/server/src/main/java/com/project/application/models/exercise/MuscleGroupDetailsResource.java
package com.project.application.models.exercise;

import com.project.domain.workout.MuscleGroup;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record MuscleGroupDetailsResource(String id, String name, String translatedName) {
    public static MuscleGroupDetailsResource from(MuscleGroup muscleGroup) {
        return new MuscleGroupDetailsResource(muscleGroup.getId(), muscleGroup.getName(), muscleGroup.getTranslatedName());
    }
}

File: src/server/src/main/kotlin/com/project/application/models/exercise/MuscleGroupDetailsResource.kt
package com.project.application.models.exercise

import com.project.domain.workout.MuscleGroup
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class MuscleGroupDetailsResource(val id: String?, val name: String?, val translatedName: String?) {
    companion object {
        fun from(muscleGroup: MuscleGroup): MuscleGroupDetailsResource {
            return MuscleGroupDetailsResource(muscleGroup.id, muscleGroup.name, muscleGroup.translatedName)
        }
    }
}
